systemd/systemd.go

```diff
@@ -972,14 +988,14 @@ func (s *systemd) Kill(serviceName, signal, who string) error {
 	return err
 }
 
-func (s *systemd) Restart(serviceName string, timeout time.Duration) error {
+func (s *systemd) Restart(timeout time.Duration, serviceName ...string) error {
 	if s.mode == GlobalUserMode {
 		panic("cannot call restart with GlobalUserMode")
 	}
-	if err := s.Stop(serviceName, timeout); err != nil {
+	if err := s.Stop(timeout, serviceName...); err != nil {
 		return err
 	}
-	return s.Start(serviceName)
+	return s.Start(serviceName...)
```

Comment:
I have https://github.com/snapcore/snapd/pull/10166 which makes this a single restart call.